keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlDeliveryConfigRepository.kt

```diff
@@ -1064,6 +1075,22 @@ class SqlDeliveryConfigRepository(
     }
   }
 
+  override fun markCheckComplete(deliveryConfig: DeliveryConfig) {
+    sqlRetry.withRetry(WRITE) {
+      jooq
+        .update(DELIVERY_CONFIG_LAST_CHECKED)
+        .setNull(DELIVERY_CONFIG_LAST_CHECKED.LOCKED_BY)
+        .setNull(DELIVERY_CONFIG_LAST_CHECKED.LOCKED_AT)
+        .set(DELIVERY_CONFIG_LAST_CHECKED.AT, clock.instant().toTimestamp())
+        .where(DELIVERY_CONFIG_LAST_CHECKED.DELIVERY_CONFIG_UID.eq(
+          select(DELIVERY_CONFIG.UID)
+            .from(DELIVERY_CONFIG)
+            .where(DELIVERY_CONFIG.NAME.eq(deliveryConfig.name))
+        ))
+        .execute()
+    }
+  }
```

Comment:
On completion of the check cycle clear the lock and update the last checked timestamp. Only doing this _after_ the check itself is complete means we can delay resource checks until environment checks are up-to-date.

Reply:
What's the hazard (dangerous scenario) that the lock prevents? (i.e., what is the bad thing that might happen if this code just set the last checked timestamp and didn't do any locking?)

Reply:
Multiple Keel instances trying to check the same set of environments at once. Unlikely to cause an issue, but especially if the app has been paused for a while, it's probably going to be a slower check and may delay other work.

Reply:
Just to confirm: in the unlikely event that the instance holding the lock crashes just before reaching that `finally` block where the call to this method is made, the lock will eventually expire, which will cause the environment to be rechecked, correct?

Reply:
Right, that's why the select is predicated on `locked_by` being `null` _or_ `locked_at` being <= some cutoff.